### PR TITLE
plugin: fixSomeChineseBug

### PR DESCRIPTION
**Pull request by tssailzz8**
fix https://github.com/quisquous/cactbot/issues/4287

### CODE REVIEW

**Comment by tssailzz8** [comment]
I don't know how to solve this dependency

File: plugin/CactbotEventSource/CactbotEventSource.cs [hunk 1]
```diff
@@ -268,14 +268,31 @@ public override void SaveConfig(IPluginConfig config)
       // Log this for now as there will likely be a lot of questions, re: user directories.
       if (Config.UserConfigFile != null)
         LogInfo(Strings.CactbotUserDirectory, Config.UserConfigFile);
-
+      var gameRegion = Machina.FFXIV.Headers.Opcodes.OpcodeManager.Instance.GameRegion;
```
Comment (quisquous) [inline plugin/CactbotEventSource/CactbotEventSource.cs:271]: Is there a reason to check the language below? Should this always use only the game region instead of game region + language?

File: plugin/CactbotEventSource/CactbotEventSource.csproj [hunk 2]
```diff
@@ -52,28 +52,28 @@
   </PropertyGroup>
   <ItemGroup>
     <Reference Include="Advanced Combat Tracker">
-      <HintPath>..\ThirdParty\ACT\Advanced Combat Tracker.exe</HintPath>
+      <HintPath>E:\Advanced Combat Tracker\Advanced Combat Tracker.exe</HintPath>
```
Comment (quisquous) [inline plugin/CactbotEventSource/CactbotEventSource.csproj:55]: And as Akurosia says, you should revert all the changes to csproj files to fix the build errors.

File: plugin/CactbotEventSource/CactbotEventSource.cs (hunk 1 above)
Comment (tssailzz8) [reply plugin/CactbotEventSource/CactbotEventSource.cs]: This is to load different opcode

Comment (tssailzz8) [reply plugin/CactbotEventSource/CactbotEventSource.cs]: it can also ues in ngld https://github.com/ngld/OverlayPlugin/blob/1a4232e189204c15f07a57cdeb48afa8b6c37990/OverlayPlugin.Core/EventSources/EnmityEventSource.cs#L73

Comment (quisquous) [reply plugin/CactbotEventSource/CactbotEventSource.cs]: Sorry, I was trying to suggest something more like:

```cs
if (gameRegion==Machina.FFXIV.GameRegion.Chinese) {
  ffxiv_ = new FFXIVProcessCn(this);
} else if (gameRegion==Machina.FFXIV.GameRegion.Korean) {
  ffxiv_ = new FFXIVProcessKo(this);
} else {
  ffxiv_ = new FFXIVProcessIntl(this);
}
```

...without checking the language setting in the ffxiv act plugin.  If we have the game region, I don't think the language matters anymore.  Does that make sense?

File: plugin/CactbotOverlay/CactbotOverlay.csproj [hunk 3]
```diff
@@ -50,17 +50,14 @@
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
   </PropertyGroup>
   <ItemGroup>
-    <Reference Include="Advanced Combat Tracker, Version=3.4.5.266, Culture=neutral, PublicKeyToken=a946b61e93d97868, processorArchitecture=MSIL">
-      <SpecificVersion>False</SpecificVersion>
-      <HintPath>..\ThirdParty\ACT\Advanced Combat Tracker.exe</HintPath>
+    <Reference Include="Advanced Combat Tracker">
+      <HintPath>E:\Advanced Combat Tracker\Advanced Combat Tracker.exe</HintPath>
```
Comment (quisquous) [inline plugin/CactbotOverlay/CactbotOverlay.csproj:54]: Can you also revert these changes to this csproj file?

**Comment by tssailzz8** [comment]
in the first it is rely on Machina.FFXIV.dll https://github.com/ravahn/machina
so i use reflection to not use it

File: plugin/CactbotOverlay/CactbotOverlay.csproj (hunk 3 above)
Comment (quisquous) [reply plugin/CactbotOverlay/CactbotOverlay.csproj:54]: Sorry, I don't think you should change the csproj.  Can you revert your changes to this file?

**Comment by tssailzz8** [comment]
i am not good at git,maybe  i pr it again

**Comment by quisquous** [comment]
Sorry! You should be able to do: `git checkout main plugin/CactbotOverlay/CactbotOverlay.csproj`
